### DISCUSSION TITLE
Possibility to export (XAML) view in PDF

### DISCUSSION THREAD

**Discussion by alessandro-gandolfi-ors** [AvaloniaUI/Avalonia#13077] [Ideas]
Technically this feature was possible in versions before 11, since then a lot of classes can't be accessed and it's not possible without using reflection.

This could be useful when in a program someone would like to export charts or other dinamically rendered content.

With the headless platform, it could be even used to create a service to create a report (which is my goal).

It was previously possible with WPF but development and support are stale and it is limited to Windows, this could be a great opportunity for Avalonia to grow.

The 2 classes that could make this possible if made public again are DrawingContext and ImmediateRender as found in the example of https://github.com/Oaz/AvaloniaUI.PrintToPDF. Other implementations are also welcome if access to those classes has to remain as it is now.

**Comment by BreyerW** [comment]
Instead of directly doing PDF consider rendering to bitmap and - if necessary - draw bitmap to pdf. If pdf isnt absolutely necessary you can save bitmap directly as image file, somewhat like here [https://github.com/AvaloniaUI/Avalonia.HtmlRenderer/issues/10](https://github.com/AvaloniaUI/Avalonia.HtmlRenderer/issues/10)

**Reply by alessandro-gandolfi-ors** [reply]
I don't think it would be an equal option to render directly to PDF. Text selection and search wouldn't be possible as the PDF would be generated by a bunch of images. In contexts as mine where a lot of data is displayed, those features are useful.

**Comment by maxkatz6** [comment]
See https://github.com/AvaloniaUI/Avalonia/pull/13128, most likely `Avalonia 11.1 ` will have this extension method shipped. With which you can use the same approach as in AvaloniaUI.PrintToPDF.

**Reply by timunie** [reply]
as this is going to be merged soon, nightly should also have it in latest tomorrow, if you are curious to try it out :-)

**Reply by alessandro-gandolfi-ors** [reply]
I will try it asap! Thanks a lot for the support!
